src/main/java/org/jenkinsci/main/modules/instance_identity/InstanceIdentity.java

```diff
@@ -59,6 +59,12 @@ public InstanceIdentity(File keyFile) throws IOException {
             try {
                 in = new StringReader(new String(KEY.decrypt().doFinal(enc), "UTF-8"));
             } catch (GeneralSecurityException x) {
+                LOGGER.log(Level.SEVERE, "identity.key.enc is corrupted. Deleting identity.key.enc and generating a new one");
+                Util.deleteFile(keyFile);
+                gen.initialize(2048, new SecureRandom()); // going beyond 2048 requires crypto extension
+                keys = gen.generateKeyPair();
+                write(keys, keyFile);
+                LOGGER.log(Level.SEVERE, "New identity.key.enc was successfully created");
```

Comment:
Presumably _this_ message is not `SEVERE`. Probably it could be skipped altogether; we have already said we were going to do it.